ci/scripts/r_test.sh

```diff
@@ -24,16 +24,27 @@ source_dir=${1}/r
 
 pushd ${source_dir}
 
-export LD_LIBRARY_PATH=${ARROW_HOME}/lib:${LD_LIBRARY_PATH}
-export R_LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
-export ARROW_R_CXXFLAGS='-Werror'
+if [ "$ARROW_USE_PKG_CONFIG" != "FALSE" ]; then
+  export LD_LIBRARY_PATH=${ARROW_HOME}/lib:${LD_LIBRARY_PATH}
+  export R_LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+fi
+if [ "$ARROW_R_CXXFLAGS" != "" ]; then
+  export _R_CHECK_COMPILATION_FLAGS_=FALSE
+fi
 export TEST_R_WITH_ARROW=TRUE
 export _R_CHECK_TESTS_NLINES_=0
 export _R_CHECK_CRAN_INCOMING_REMOTE_=FALSE
 export _R_CHECK_LIMIT_CORES_=FALSE
-export _R_CHECK_COMPILATION_FLAGS_=FALSE
 export VERSION=$(grep ^Version DESCRIPTION | sed s/Version:\ //)
 
+# Make sure we aren't writing to the home dir (CRAN _hates_ this but there is no official check)
+BEFORE=$(ls -alh ~/)
+
 ${R_BIN} -e "rcmdcheck::rcmdcheck(build_args = '--no-build-vignettes', args = c('--no-manual', '--as-cran', '--ignore-vignettes', '--run-donttest'), error_on = 'warning', check_dir = 'check')"
 
+AFTER=$(ls -alh ~/)
+if [ "$BEFORE" != "$AFTER" ]; then
```

Comment:
This will fail if `CMAKE_EXPORT_NO_PACKAGE_REGISTRY` isn't set, right? e.g. if a developer runs docker-compose directly to test some R changes.

Reply:
There are two scenarios where this script is run:

1. `docker-compose run ubuntu-r`, in which you've already built the C++ library so cmake isn't invoked by `R CMD check`
2. `docker-compose run r`, which uses the `r/inst/build_arrow_static.sh` script, which calls cmake with `CMAKE_EXPORT_NO_PACKAGE_REGISTRY` set

Both should be safe for local dev. I confirmed the latter but the former (which you're probably more interested in) logically can't be a problem.